workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/InputSpecPostValidator.java

```diff
@@ -0,0 +1,85 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.validator.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.validator.Validator;
+import io.hyscale.controller.exception.ControllerErrorCodes;
+import io.hyscale.controller.model.WorkflowContext;
+
+@Component
+public class InputSpecPostValidator implements Validator<List<WorkflowContext>> {
+
+    private static final Logger logger = LoggerFactory.getLogger(InputSpecPostValidator.class);
+
+    private List<Validator<WorkflowContext>> validators = new ArrayList<Validator<WorkflowContext>>();
+
+    public void addValidator(Validator<WorkflowContext> validator) {
+        if (validator != null) {
+            validators.add(validator);
+        }
+    }
+
+    @Override
+    public boolean validate(List<WorkflowContext> contextList) throws HyscaleException {
+        boolean isInvalid = false;
+        boolean isFailed = false;
+        StringBuilder exceptionMsg = new StringBuilder();
+        for (Validator<WorkflowContext> validator : validators) {
+            logger.debug("Running validator: {}", validator.getClass());
+            if (contextList == null) {
+                try {
+                    isInvalid = validator.validate(null) ? isInvalid : true;
+                } catch (HyscaleException e) {
+                    isFailed = true;
+                    exceptionMsg.append(e.getMessage()).append("\n");
+                }
+            } else {
+                for (WorkflowContext context : contextList) {
+                    try {
+                        isInvalid = validator.validate(context) ? isInvalid : true;
+                    } catch (HyscaleException e) {
+                        isFailed = true;
+                        exceptionMsg.append(e.getMessage()).append("\n");
+                    }
+                }
+            }
+            if (isInvalid || isFailed) {
+                logger.error("Input invalid : {}, failed: {}, error message : {}", isInvalid, isFailed,
+                        exceptionMsg.toString());
+                WorkflowLogger.logPersistedActivities();
+            }
+            if (isFailed) {
+                throw new HyscaleException(ControllerErrorCodes.INPUT_VALIDATION_FAILED,
+                        ToolConstants.INVALID_INPUT_ERROR_CODE, exceptionMsg.toString());
```

Comment:
isInvalid and isFailed any difference in terms of user error message

Reply:
Failed covers Exception cases and Invalid covers validation failures. The message would be validator dependent. But here only the processing is different.